mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -175,6 +177,33 @@ class VectorsSuite extends FunSuite {
     assert(v.size === x.rows)
   }
 
+  test("sqdist") {
+    val random = new Random(System.nanoTime())
+    for (m <- 1 until 1000 by 10) {
+      val nnz = random.nextInt(m) + 1
+
+      val indices1 = random.shuffle(0 to m - 1).toArray.slice(0, nnz).sorted
```

Comment:
Do you mean to write ```0 until m``` here?  If so, it could fail since it would produce an invalid index.